File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/UsedeskFileInfo.java
package ru.usedesk.chat_sdk.external.entity;

import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class UsedeskFileInfo {
    private final Uri uri;
    private final Type type;

    public UsedeskFileInfo(@NonNull Uri uri, @NonNull Type type) {
        this.uri = uri;
        this.type = type;
    }

    @NonNull
    public Uri getUri() {
        return uri;
    }

    @NonNull
    public Type getType() {
        return type;
    }

    public enum Type {
        IMAGE,
        VIDEO,
        DOCUMENT,
        OTHER;

        public static Type getByMimeType(@Nullable String mimeType) {
            if (mimeType != null) {
                if (mimeType.startsWith("image/")) {
                    return IMAGE;
                } else if (mimeType.startsWith("video/")) {
                    return VIDEO;
                } else if (mimeType.startsWith("doc/")) {
                    return DOCUMENT;
                }
            }
            return OTHER;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/entity/UsedeskFileInfo.kt

package ru.usedesk.chat_sdk.entity

import android.content.Context
import android.net.Uri
import ru.usedesk.common_sdk.utils.UsedeskFileUtil.getFileName
import ru.usedesk.common_sdk.utils.UsedeskFileUtil.getMimeType

data class UsedeskFileInfo(
    val uri: Uri,
    val type: String,
    val name: String
) {

    fun isImage(): Boolean = type.startsWith(IMAGE_TYPE)

    fun isVideo(): Boolean = type.startsWith(VIDEO_TYPE)

    fun isAudio(): Boolean = type.startsWith(AUDIO_TYPE)

    companion object {
        const val IMAGE_TYPE = "image/"
        const val VIDEO_TYPE = "video/"
        const val AUDIO_TYPE = "audio/"

        fun create(context: Context, uri: Uri): UsedeskFileInfo {
            val mimeType = context.getMimeType(uri)
            val name = context.getFileName(uri)
            return UsedeskFileInfo(uri, mimeType, name)
        }
    }
}